### PR TITLE
Add GCC7 jobs to Travis

### PR DESCRIPTION
**Pull request by 4tXJ7f**
This commit adds two jobs (debug, with portfolio, test groups 0 and 1) that are
allowed to fail to Travis. The jobs are allowed to fail because CVC4 has known
problems with GCC7 (see issue #1050). Both jobs are added using matrix.include,
based on the example in the documentation:
https://docs.travis-ci.com/user/languages/cpp/#GCC-on-Linux . This
unfortunately requires some code duplication but there does not seem to be a
way to do it in a much better fashion.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -106,8 +110,31 @@ script:
...
+    - os: linux
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - *common_deps
+            - g++-7
+      env:
+        - MATRIX_EVAL='CC=gcc-7 && CXX=g++-7'
```
Comment (mpreiner) [inline .travis.yml]: I think you can use global and matrix here to get rid of the TEST_GROUP=1 rule.

Something like that (not tested):

env:
  global:
    - MATRIX_EVAL='CC=gcc-7 && CXX=g++-7'
    - TRAVIS_CVC4=yes TRAVIS_CVC4_CHECK_PORTFOLIO=yes TRAVIS_WITH_LFSC=yes TRAVIS_CVC4_JAVA_API_TEST=yes TRAVIS_CVC4_CONFIG='debug --enable-language-bindings=java,c --with-lfsc --with-portfolio'
  matrix:
    - TEST_GROUP=0
    - TEST_GROUP=1

File: .travis.yml [hunk 2]
```diff
@@ -137,6 +157,14 @@ matrix:
           branch_pattern: coverity_scan
       after_failure:
         - cat /home/travis/build/CVC4/CVC4/cov-int/build-log.txt
+  allow_failures:
+  - env:
```
Comment (mpreiner) [inline .travis.yml]: Same as above with global/matrix.

**Comment by 4tXJ7f** [comment]
Hmm, it looks like the nested global/matrix stuff doesn't really work, unfortunately: https://travis-ci.org/CVC4/CVC4/builds/269285208?utm_source=github_status&utm_medium=notification

File: .travis.yml [hunk 3]
```diff
@@ -124,6 +152,14 @@ matrix:
...
+    - MATRIX_EVAL='CC=gcc-7 && CXX=g++-7'
+    - TRAVIS_CVC4=yes TRAVIS_CVC4_CHECK_PORTFOLIO=yes TRAVIS_WITH_LFSC=yes TRAVIS_CVC4_JAVA_API_TEST=yes TRAVIS_CVC4_CONFIG='debug --enable-language-bindings=java,c --with-lfsc --with-portfolio' TEST_GROUP=0
+  - env:
+    - MATRIX_EVAL='CC=gcc-7 && CXX=g++-7'
+    - TRAVIS_CVC4=yes TRAVIS_CVC4_CHECK_PORTFOLIO=yes TRAVIS_WITH_LFSC=yes TRAVIS_CVC4_CONFIG='debug --enable-language-bindings=java,c --with-lfsc --with-portfolio' TEST_GROUP=1
+
```
Comment (timothy-king) [inline .travis.yml]: These are going to be really brittle. Do we want to fix the underlying problems first, and not have allow_failures?

Comment (4tXJ7f) [reply .travis.yml]: The idea was to have these jobs to easily test any incoming fixes of #1050 and then to immediately remove the allowed failures but I definitely agree that they are brittle.

**Comment by aniemetz** [comment]
We have to wait with merging this PR until Andy's PR is merged in, of course.

**Pull request merged**